### PR TITLE
DorresteinLab Chromatogram Pairing PR 

### PR DESCRIPTION
**Pull request by abisarvepalli**

### CODE REVIEW

**Comment by tomas-pluskal** [comment]
Hi,
Looks much better now!
I see the PR contains a change to src/main/lib/dulab/adap/3.2.0/adap-3.2.0.jar, please check that.

**Comment by tomas-pluskal** [comment]
Also, the Travis build check failed. Please take a look at the error log.

**Comment by abisarvepalli** [comment]
Checking right now!

**Comment by abisarvepalli** [comment]
Just fixed the .jar file

File: src/main/java/net/sf/mzmine/datamodel/impl/SimplePeakIdentity.java [hunk 1]
```diff
@@ -131,9 +131,9 @@ public String toString() {
     }
 
     @Override
-    // Removed @Nonnull in front because the function may return null
-    public String getPropertyValue(final String property) {
-		return properties.get(property);
+    public @Nonnull String getPropertyValue(final String property) {
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/datamodel/impl/SimplePeakIdentity.java]: This change should not be here

File: src/main/java/net/sf/mzmine/modules/masslistmethods/chromatogrambuilder/ChromatogramBuilderTask.java [hunk 2]
```diff
@@ -191,6 +211,7 @@ public void run() {
         }
 
         // Add new peaklist to the project
+        System.out.print(this.RTRangeMSMS);
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/modules/masslistmethods/chromatogrambuilder/ChromatogramBuilderTask.java]: pls use the logger here

**Comment by tomas-pluskal** [comment]
A question: you added two new parameters to chromatogram builder, but I think these parameters would actually make more sense in the chromatogram deconvolution module. Because that is the module, which ultimately decides the MS/MS spectrum for a peak. And you also use the saved values from chromatogram builder in there - however, it won't work if the user uses the ADAP chromatogram builder, for example.

**Comment by tomas-pluskal** [comment]
Adding the getMZrangeMSMS and getRTrangeMSMS methods to all Feature implementing classes seems unnecessary - you wouldn't need that if you added the parameters to the chromatogram deconvolution module.

File: src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/manual/ManualPeak.java [hunk 3]
```diff
@@ -309,4 +309,16 @@ public SimplePeakInformation getPeakInformation(){
     }
     //End dulab Edit
 
+	@Override
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/manual/ManualPeak.java]: I guess these edits to ManualPeak are not needed anymore?

Comment (abisarvepalli) [inline src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/manual/ManualPeak.java]: Ya getting rid of that right now. It was the reason for the syntax errors that caused errors in the checks.

**Comment by abisarvepalli** [comment]
With the omission of the override tags, all checks have passed and think that this is ready!

**Comment by tomas-pluskal** [comment]
The PR contains some changes to adap3decompositionV1_5/help/help.html
Did you make those changes, or are you reverting somebody else's commit?

**Comment by abisarvepalli** [comment]
Seems that we were reverting somebody else's commit and that issue has been cleared.

**Comment by tomas-pluskal** [comment]
OK, looks good.

**Comment by lfnothias** [comment]
It's good to hear ! Now it's good to go on our side. Thank you very much for your help and your patience.

**Pull request merged**